src/CabalHelper/Runtime/Main.hs

```diff
@@ -373,13 +398,33 @@ main = do
       return $ Just $ ChResponsePkgDbs $ map pkgDb $ withPackageDB lbi
 
     "entrypoints":[] -> do
+#if CH_MIN_VERSION_Cabal(2,0,0)
+      includeDirMap <- recursiveDepInfo lbi v distdir
+      eps <- componentsMap lbi v distdir $ \c clbi _bi -> do
```

Comment:
We could remove the `do` here since both cases are `return`.

Reply:
Indeed, we can also pull the returns out of the two cases.

Reply:
Except `componentsMap` is in IO, so requires a `return`.